cmd/protecodeExecuteScan.go

```diff
@@ -0,0 +1,317 @@
+	c := command.Command{}
+	// reroute command output to loging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	return runProtecodeScan(&config, influx)
+}
+
+func runProtecodeScan(config *protecodeExecuteScanOptions, influx *protecodeExecuteScanInflux) error {
+
+	//create client for sending api request
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, create protecode client")
+	}
+	client := createClient(config)
+
+	if config.Verbose {
+		log.Entry().Infof("Protecode scan debug, get docker image: %v, %v, %v, %v", config.ScanImage, config.DockerRegistryURL, config.FilePath, config.IncludeLayers)
+	}
+	fileName, filePath := getDockerImage(config.ScanImage, config.DockerRegistryURL, config.FilePath, config.IncludeLayers)
+	if len(config.FilePath) <= 0 {
+		(*config).FilePath = filePath
+		if config.Verbose {
+			log.Entry().Infof("Protecode scan debug, filepath: %v", config.FilePath)
+		}
+	}
+
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, execute protecode scan")
+	}
+	parsedResult, productID := executeProtecodeScan(client, config, fileName, writeReportToFile)
+
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, write influx data")
+	}
+	setInfluxData(influx, parsedResult)
+
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, write report to filesystem")
+	}
+	writeReportDataToJSONFile(config, parsedResult, productID, ioutil.WriteFile)
+
+	err := os.Remove(config.FilePath)
+	if err != nil {
+		log.Entry().WithError(err).Warnf("Protecode scan warning, failed to delete tar source code")
+	}
+
+	return nil
+}
+
+var getImage = func(scanImage string, registryURL string, includeLayers bool, cacheImagePath string) pkgutil.Image {
+
+	completeURL := getURLAndFileNameFromDockerImage(scanImage, registryURL)
+	image, err := pkgutil.GetImage(completeURL, includeLayers, cacheImagePath)
+	if err != nil {
+		log.Entry().Fatalf("Protecode scan failed, exception during get docker image %v", err)
+	}
+
+	return image
+}
+
+func getDockerImage(scanImage string, registryURL string, path string, includeLayers bool) (string, string) {
+	fileName := scanImage
+
+	cachePath := "./cache"
+	cacheProtecodeImagePath := "/protecodeImage"
+	cacheImagePath := filepath.Join(cachePath, cacheProtecodeImagePath)
+	os.Mkdir(cacheImagePath, 600)
+
+	image := getImage(scanImage, registryURL, includeLayers, cacheImagePath)
+
+	tarFileName := filepath.Join(cacheImagePath, fileName)
+
+	if !pkgutil.IsTar(scanImage) {
+
+		fileName = fmt.Sprintf("%v.tar", strings.ReplaceAll(scanImage, "/", "_"))
+		tarFileName = filepath.Join(cachePath, fileName)
+
+		tarImageData(tarFileName, image, cacheImagePath)
+	}
+
+	var resultFilePath string
+
+	if len(path) <= 0 {
+		resultFilePath = filepath.Join(cachePath, fileName)
+		if len(resultFilePath) <= 0 {
+			log.Entry().Fatalf("Protecode scan failed, there is no file path configured  : %v (filename:%v, PSPath: %v)", path, fileName, image.FSPath)
```

Comment:
minor: not sure if we need `Protecode can failed` in the message since we always add the step name to the log as field ...